pkg/scheduler/schedulerpb/scheduler.proto

```diff
@@ -72,6 +72,15 @@ enum FrontendToSchedulerType {
   CANCEL = 2;
 }
 
+enum QueryComponent {
+  UNSPECIFIED = 0; // fallback in case of some failure?
+  INGESTERS_ONLY = 1;
+  STORE_GATEWAYS_ONLY = 2;
+  // expressing combinations like this is seems kind of awkwardly overly-specific and brittle
+  // but I can't really see this needing more values anytime soon and gRPC is easy-ish to change
+  INGESTERS_AND_STORE_GATEWAYS = 3;
```

Comment:
could also be like `ALL` if we don't want a concept as specific as `INGESTERS_AND_STORE_GATEWAYS`